detox/android/detox/src/full/java/com/wix/detox/reactnative/idlingresources/UIModuleIdlingResource.java

```diff
@@ -122,6 +140,33 @@ protected boolean checkIdle() {
         return true;
     }
 
+    // This is a workaround for https://github.com/facebook/react-native/issues/32594
+    // uses duck typing heuristics to determine that this is probably the stuck Switch operation and if so, ignores it
+    private boolean workaroundForRN66Bug(Object uiOperationQueue) {
+        boolean isStuckSwitchOperation = false;
+
+        try {
+            Class<?> DispatchCommandViewOperation = Class.forName(CLASS_DISPATCH_COMMAND_VIEW_OPERATION);
+            List<?> mViewCommandOperations = Reflect.on(uiOperationQueue).field(FIELD_MVIEW_COMMAND_OPERATIONS).get();
+
+            if (mViewCommandOperations.size() == 1) {
+                Class<?> ReadableArray = Class.forName(CLASS_READABLE_ARRAY);
+                Object viewOperation = DispatchCommandViewOperation.cast(mViewCommandOperations.get(0));
+
+                Object viewArgs = ReadableArray.cast(Reflect.on(viewOperation).field(FIELD_MARGS).get());
+                Method getReadableArrayBoolean = viewArgs.getClass().getDeclaredMethod(METHOD_GET_BOOLEAN, int.class);
+                getReadableArrayBoolean.setAccessible(true);
+                Boolean nativeValueArg = (Boolean) getReadableArrayBoolean.invoke(viewArgs, 0);
+
+                int numRetries = Reflect.on(viewOperation).field(FIELD_NUM_RETRIES).get();
+                String mCommand = Reflect.on(viewOperation).field(FIELD_MCOMMAND).get();
+
+                isStuckSwitchOperation = (numRetries == 1 && mCommand.equals(SET_NATIVE_VALUE) && nativeValueArg != null);
```

Comment:
So this is pretty good! 💪🏻💪🏻💪🏻  But since false-idles is an insanely sensitive issue, I must ask for two improvements over it:

1. I noticed that using the operation's `mTag` field, there is a way to reach the actual Android view instance (!), provided we can get a hold of the `NativeViewHierarchyManager` instance: property `NativeViewHierarchyManager.mTagsToViews` is a `tag-->view` map. Ultimately, we need to assert that the view's class type is indeed the naughty switch that's causing all the issues.
1. Let's not conclude "idle" immediately in the first occurrence of things. Rather, since as you said -- the operation will be stuck forever, we can double check that this isn't just a bad coincidence (and thus avoid a false-idle...) by stabilizing the check such that we'd only declare idle if all of the above conditions are met X times in row (X can be even as high a 10, IMO).

With that, I'd get some better sleep at night 😄